components/dash-core-components/tests/integration/graph/test_graph_varia.py

```diff
@@ -847,3 +859,77 @@ def graph_dims():
     assert graph_dims() == responsive_size
 
     assert dash_dcc.get_logs() == []
+
+
+def test_grva010_external_mathjax_prevents_lazy(dash_dcc):
+    app = Dash(
+        __name__,
+        eager_loading=False,
+        external_scripts=["https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-svg.js"],
+    )
+
+    app.layout = html.Div(id="div", children=[html.Button(id="btn")])
+
+    @app.callback(Output("div", "children"), [Input("btn", "n_clicks")])
+    def load_chart(n_clicks):
+        if n_clicks is None:
+            raise PreventUpdate
+
+        return dcc.Graph(mathjax=True, id="output", figure={"data": [{"y": [3, 1, 2]}]})
+
+    dash_dcc.start_server(
+        app,
+        debug=True,
+        use_reloader=False,
+        use_debugger=True,
+        dev_tools_hot_reload=False,
+    )
+
+    # Give time for the async dependency to be requested (if any)
+    dash_dcc.wait_for_element("button#btn")
+
+    scripts = dash_dcc.driver.find_elements(By.CSS_SELECTOR, "script")
+    assert findSyncMathJax(scripts) is None
+    assert findAsyncMathJax(scripts) is None
+
+    dash_dcc.find_element("#btn").click()
+
+    # Give time for the async dependency to be requested (if any)
+    time.sleep(2)
+
+    scripts = dash_dcc.driver.find_elements(By.CSS_SELECTOR, "script")
+    assert findSyncMathJax(scripts) is None
+    assert findAsyncMathJax(scripts) is None
+
+    assert dash_dcc.get_logs() == []
+
+
+@pytest.mark.parametrize("is_eager", [True, False])
+def test_grva011_without_mathjax(dash_dcc, is_eager):
+    app = Dash(__name__, eager_loading=is_eager, assets_folder="../../assets")
+
+    app.layout = html.Div([dcc.Graph(id="output", figure={"data": [{"y": [3, 1, 2]}]})])
+
+    dash_dcc.start_server(app)
+    assert dash_dcc.get_logs() == []
```

Comment:
Here we should insert something into the graph that WOULD be rendered as math if that were enabled, and verify that it is left as normal text. So if using the title I suggested for the previous test, the assertion could be like:
```py
assert dash_dcc.wait_for_element(".gtitle").text == "$E=mc^2$"
```
If it's rendered as math, that text is `""` because it's just a bunch of SVG paths.

Also we should assert that `find(Sync|Async)MathJax` is as expected - I guess `sync==is_eager`, `async==False`?

Reply:
Addressed in 6623ba2ddf93efbc8e43a89d06c02b67f0e77c2c.

Reply:
In respect to `find(Sync|Async)MathJax`, I added assertions in 980961f99a2211a4c72bc390d45a34863a3eb170.
But please verify the cases are actually correct?

Reply:
Ah those assertions are a bit off, but also maybe not really useful: In the `eager_loading=True` case I see:
```
<script src="/_dash-component-suites/dash/dcc/async-mathjax.v2_2_0m1646920173.js"></script>
```
It's still using `async-` because unlike `plotly.js` we use the same bundle for the sync and async versions. TBH I don't recall why we did that for plotly.js, here it should be fine to use the same. But then in the default `eager_loading=False` case I see no MathJax script. I had forgotten about this, but it's because Webpack for some reason recently started deleting its async script tags after they finish loading. https://github.com/plotly/dash/pull/1685#issuecomment-877199466

So actually let's forget about the script tags entirely, and just make assertions about `!!window.MathJax`. That's what we ultimately care about anyway.

Reply:
https://github.com/plotly/dash/pull/1949/commits/70af21e8cf96bab4627246ea1327d7534baa09e1